lib/datadog/core/remote/client.rb

```diff
@@ -185,8 +194,35 @@ def capabilities_binary
         def register_receivers
           matcher = Dispatcher::Matcher::Product.new(products)
 
-          dispatcher.receivers << Dispatcher::Receiver.new(matcher) do |_repository, changes|
-            changes.each { |change| Datadog.logger.debug { "remote config change: #{change.path.inspect}" } }
+          dispatcher.receivers << Dispatcher::Receiver.new(matcher) do |repository, changes|
+            changes.each do |change|
+              Datadog.logger.debug { "remote config change: '#{change.path}'" }
+            end
+
+            rules = repository.contents.select do |content|
+              content.path.product == 'ASM_DD'
+            end.map! { |content| JSON.parse(content.data.read.tap { content.data.rewind }) }
+
+            data = repository.contents.select do |content|
+              content.path.product == 'ASM_DATA' && ['blocked_ips', 'blocked_users' ].include?(content.path.config_id)
+            end.map! { |content| JSON.parse(content.data.read.tap { content.data.rewind }) }
+
+            overrides = repository.contents.select do |content|
+              content.path.product == 'ASM' && ['blocking', 'disabled_rules'].include?(content.path.config_id)
+            end.map! { |content| JSON.parse(content.data.read.tap { content.data.rewind }) }
+
+            exclusions = repository.contents.select do |content|
+              content.path.product == 'ASM' && content.path.config_id == 'exclusion_filters'
+            end.map! { |content| JSON.parse(content.data.read.tap { content.data.rewind }) }
```

Comment:
All of the calls to `.map!` could break if the sync does not include those products. Should we hoist the result into an intermediate variable and check for it before?

Reply:
The calls to `map!` themselves won't break: `contents.select` would just return an empty array.

What should be done here is handling the corner case of having these variables be empty, in which case we should fill them with the default local values (e.g vendored ruleset).

Reply:
> The calls to map! themselves won't break: contents.select would just return an empty array.

You are right 😄